src/main/java/org/apache/commons/crypto/stream/CryptoOutputStream.java

```diff
@@ -298,30 +301,47 @@ public boolean isOpen() {
     @Override
     public int write(ByteBuffer src) throws IOException {
         checkStream();
-        final int len = src.remaining();
-        int remaining = len;
+
+        if (transferInProgress && outBuffer.hasRemaining()) {
+            output.write(outBuffer);
+            return 0;
```

Comment:
One improvement here would be to not return if all of `outBuffer` has been written, in which case it might still be possible to write to the output.